runtime/common/src/parachains.rs

```diff
@@ -1509,7 +1541,13 @@ impl<T: Trait> ProvideInherent for Module<T> {
 			.expect("Parachain heads could not be decoded.")
 			.expect("No parachain heads found in inherent data.");
 
-		Some(Call::set_heads(data))
+		// Temporary solution for:
+		// https://github.com/paritytech/polkadot/issues/1327
+		if Self::check_candidates_signatures(&data).is_ok() {
```

Comment:
Could this not just do `Module::<T>::set_heads(Origin::none(), ...).is_ok()` without the other changes?

Reply:
Yes this would work as well. But set heads is doing much more and we don't really need this here IMHO. This just wastes resources.